### PR TITLE
Return closures from conditions

### PR DESCRIPTION
**Pull request by Shatur**
In Bevy all conditions done this way. See the [the example](https://github.com/bevyengine/bevy/blob/7fe08535dfe99fb2ff999009bb497db61b5196ec/examples/ecs/run_conditions.rs#L81). Let's make them consistent with the rest of the engine.

### CODE REVIEW

File: bevy_renet/src/transport.rs [hunk 1]
```diff
@@ -104,23 +104,51 @@ impl NetcodeClientPlugin {
...
+    |mut last_connected: Local<bool>, transport| {
+        let connected = transport.filter(|transport| transport.is_connected()).is_some();
+
+        if !*last_connected && connected {
+            *last_connected = true;
+            true
+        } else {
+            *last_connected = connected;
+            false
+        }
```
Comment (lucaspoffo) [inline bevy_renet/src/transport.rs:138]:
```suggestion
let Some(transport) = transport else {
   return false;
};

let connected = transport.is_connected();
let just_connected = !*last_connected && connected;
*last_connected = connected;
just_connected
```

Same suggestion applies for the disconnected

**Comment by lucaspoffo** [comment]
> Added two more condition that should replace #95. Similar to https://docs.rs/bevy/latest/bevy/input/struct.Input.html#method.just_pressed.

Yeah, I prefer doing like this instead of events 👍

File: bevy_renet/src/transport.rs (hunk 1 above)
Comment (Shatur) [reply bevy_renet/src/transport.rs]: I like it this way a lot, thanks!

**Comment by lucaspoffo** [comment]
Thanks!

**Pull request merged**